lib/commands/publish.js

```diff
@@ -136,6 +137,9 @@ class Publish extends BaseCommand {
     npa(`${manifest.name}@${defaultTag}`)
 
     const registry = npmFetch.pickRegistry(resolved, opts)
+
+    await oidc({ packageName: manifest.name, registry, opts, config: this.npm.config })
```

Comment:
Optional: Is there a pithy way we can indicate we're doing an OIDC publish in the the `notice` level log output? One idea:

```diff
 npm notice Tarball Contents
 npm notice 167B README.md
 npm notice 236B package.json
...
 npm notice total files: 2
 npm notice
- npm notice Publishing to https://registry.npmjs.org/ with tag latest and default access
+ npm notice Publishing to https://registry.npmjs.org/ with tag latest and default access, using OIDC
```

I would only include `using OIDC` or `as {CI Provider Name}` if the token exchange was successful